platform-dependent-components/problem-solver/cxx/sections-module/test/units/add_section_agent_test.cpp

```diff
@@ -34,127 +32,110 @@ using AddSectionTest = ScMemoryTest;
   ScAddr testActionNode = context.HelperFindBySystemIdtf("test_action_node3");
 
-  EXPECT_TRUE(utils::AgentUtils::applyAction(&context, testActionNode, WAIT_TIME));
-  EXPECT_TRUE(context.HelperCheckEdge(
-      scAgentsCommon::CoreKeynodes::question_finished_unsuccessfully, testActionNode, ScType::EdgeAccessConstPosPerm));
+  ScAction testAction = context.ConvertToAction(testActionNode);
 
-  SC_AGENT_UNREGISTER(AddSectionAgent)
+  context.SubscribeAgent<sectionsModule::AddSectionAgent>();
+
+  testAction.InitiateAndWait(WAIT_TIME);
+  EXPECT_TRUE(testAction.IsFinishedUnsuccessfully());
+  context.UnsubscribeAgent<sectionsModule::AddSectionAgent>();
 }
 
 TEST_F(AddSectionTest, add_section_invalid_parameters_3)
 {
-  ScMemoryContext & context = *m_ctx;
-  scAgentsCommon::CoreKeynodes::InitGlobal();
-  SectionsKeynodes::InitGlobal();
+  ScAgentContext & context = *m_ctx;
 
-  SC_AGENT_REGISTER(AddSectionAgent)
   ScsLoader loader;
   loader.loadScsFile(context, TEST_FILES_DIR_PATH + "test_add_section.scs");
   ScAddr testActionNode = context.HelperFindBySystemIdtf("test_action_node4");
 
-  EXPECT_TRUE(utils::AgentUtils::applyAction(&context, testActionNode, WAIT_TIME));
-  EXPECT_TRUE(context.HelperCheckEdge(
-      scAgentsCommon::CoreKeynodes::question_finished_unsuccessfully, testActionNode, ScType::EdgeAccessConstPosPerm));
+  ScAction testAction = context.ConvertToAction(testActionNode);
+
+  context.SubscribeAgent<sectionsModule::AddSectionAgent>();
 
-  SC_AGENT_UNREGISTER(AddSectionAgent)
+  testAction.InitiateAndWait(WAIT_TIME);
+  EXPECT_TRUE(testAction.IsFinishedUnsuccessfully());
+  context.UnsubscribeAgent<sectionsModule::AddSectionAgent>();
 }
 
 TEST_F(AddSectionTest, add_section_without_parent)
 {
-  ScMemoryContext & context = *m_ctx;
-  scAgentsCommon::CoreKeynodes::InitGlobal();
-  SectionsKeynodes::InitGlobal();
+  ScAgentContext & context = *m_ctx;
 
-  SC_AGENT_REGISTER(AddSectionAgent)
   ScsLoader loader;
   loader.loadScsFile(context, TEST_FILES_DIR_PATH + "test_add_section.scs");
   ScAddr testActionNode = context.HelperFindBySystemIdtf("test_action_node5");
-  ScAddr sectionNameAddr =
-      utils::IteratorUtils::getAnyByOutRelation(&context, testActionNode, scAgentsCommon::CoreKeynodes::rrel_1);
-
-  EXPECT_TRUE(utils::AgentUtils::applyAction(&context, testActionNode, WAIT_TIME));
-  ScIterator5Ptr it5 = context.Iterator5(
-      testActionNode,
-      ScType::EdgeDCommon,
-      ScType::Unknown,
-      ScType::EdgeAccessConstPosPerm,
-      scAgentsCommon::CoreKeynodes::nrel_answer);
-  EXPECT_TRUE(it5->Next());
-
-  ScIterator3Ptr it3 = context.Iterator3(it5->Get(2), ScType::EdgeAccessConstPosPerm, ScType::NodeConst);
+  ScAddr sectionNameAddr = utils::IteratorUtils::getAnyByOutRelation(&context, testActionNode, ScKeynodes::rrel_1);
+  ScAction testAction = context.ConvertToAction(testActionNode);
+
+  context.SubscribeAgent<sectionsModule::AddSectionAgent>();
+
+  testAction.InitiateAndWait(WAIT_TIME);
```

Comment:
this can also be wrapped in EXPECT_TRUE, previously there was check for action preforming